nifi-nar-bundles/nifi-azure-bundle/nifi-azure-record-sink/src/test/java/org/apache/nifi/services/azure/eventhub/TestAzureEventHubRecordSink.java

```diff
@@ -0,0 +1,162 @@
+    private static final String sasKey = "policyKey";
+    private AzureEventHubRecordSink azureEventHubRecordSink;
+    private static final String NULL_HEADER = null;
+    private static final String WRITER_IDENTIFIER = MockRecordWriter.class.getSimpleName();
+    private static final String IDENTIFIER = AzureEventHubRecordSink.class.getSimpleName();
+    private static final String ID_FIELD = "id";
+    private static final String ID_FIELD_VALUE = TestAzureEventHubRecordSink.class.getSimpleName();
+    private static final RecordSchema RECORD_SCHEMA = getRecordSchema();
+    private static final boolean SEND_ZERO_RESULTS = true;
+    @Mock
+    private EventHubProducerClient client;
+    @Mock
+    private EventDataBatch eventDataBatch;
+
+
+    @BeforeEach
+    void setRunner() throws Exception {
+        final org.apache.nifi.util.TestRunner runner = TestRunners.newTestRunner(NoOpProcessor.class);
+        runner.setValidateExpressionUsage(false);
+
+        final MockRecordWriter recordWriter = new MockRecordWriter(NULL_HEADER, false);
+        runner.addControllerService(WRITER_IDENTIFIER, recordWriter);
+        runner.enableControllerService(recordWriter);
+
+        azureEventHubRecordSink = new MockAzureEventHubRecordSink();
+        runner.addControllerService(IDENTIFIER, azureEventHubRecordSink);
+        runner.setProperty(azureEventHubRecordSink, AzureEventHubRecordSink.EVENT_HUB_NAME, eventHubName);
+        runner.setProperty(azureEventHubRecordSink, AzureEventHubRecordSink.NAMESPACE, namespaceName);
+        runner.setProperty(azureEventHubRecordSink, AzureEventHubRecordSink.SHARED_ACCESS_POLICY_KEY, sasKey);
+        runner.setProperty(azureEventHubRecordSink, AzureEventHubRecordSink.RECORD_WRITER_FACTORY, WRITER_IDENTIFIER);
+        runner.enableControllerService(azureEventHubRecordSink);
+    }
+    public class MockAzureEventHubRecordSink extends AzureEventHubRecordSink {
+
+        @Override
+        protected EventHubProducerClient createEventHubClient(
+                final String namespace,
+                final String serviceBusEndpoint,
+                final String eventHubName,
+                final String policyName,
+                final String policyKey,
+                final AzureAuthenticationStrategy authenticationStrategy) throws ProcessException {
+            return client;
+        }
+    }
+
+    private static RecordSchema getRecordSchema() {
+        final RecordField idField = new RecordField(ID_FIELD, RecordFieldType.STRING.getDataType());
+        return new SimpleRecordSchema(Collections.singletonList(idField));
+    }
+
+    private static Record[] getRecords(int numberOfRecords) {
+        final Map<String, Object> values = Collections.singletonMap(ID_FIELD, ID_FIELD_VALUE);
+        final Record record = new MapRecord(RECORD_SCHEMA, values);
+        final Record[] records = new Record[numberOfRecords];
+        Arrays.fill(records, record);
+        return records;
+    }
```

Comment:
Recommend moving the mock class and static methods to the end of the test class, after the test methods.